model/qti/ItemMaxScoreService.php

```diff
@@ -0,0 +1,172 @@
+use Exception;
+use InvalidArgumentException;
+use oat\oatbox\service\ConfigurableService;
+use common_Logger;
+use common_Utils;
+
+/**
+ * Retrieves MAXSCORE (maximum achievable points) from QTI items.
+ */
+class ItemMaxScoreService extends ConfigurableService
+{
+    public const SERVICE_ID = 'taoQtiItem/ItemMaxScoreService';
+
+    /**
+     * Get MAXSCORE for a single item
+     *
+     * Retrieves the maximum achievable score for a single item.
+     *
+     * Behavior:
+     * - Returns 0.0 for items with missing MAXSCORE outcomeDeclaration
+     * - Returns 0.0 for items that fail to parse (with error logged)
+     * - Handles external-scored items (returns manual MAXSCORE if set)
+     *
+     * @param string $itemUri - Valid TAO resource URI (e.g., 'http://tao.dev/ontology.rdf#item123')
+     * @return float
+     * @throws InvalidArgumentException if $itemUri is empty or not a valid URI
+     */
+    public function getItemMaxScore(string $itemUri): float
+    {
+        if (empty($itemUri) || !common_Utils::isUri($itemUri)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid item URI provided: "%s". Expected a valid TAO resource URI.',
+                    $itemUri
+                )
+            );
+        }
+
+        try {
+            $item = new core_kernel_classes_Resource($itemUri);
+
+            /** @var Service $qtiService */
+            $qtiService = $this->getServiceLocator()->get(Service::class);
+
+            /** @var Item $qtiItem */
+            $qtiItem = $qtiService->getDataItemByRdfItem($item);
+
+            return $this->extractMaxScore($qtiItem);
+        } catch (Exception $e) {
+            common_Logger::w(
+                'Failed to retrieve MAXSCORE for item ' . $itemUri . ': ' . $e->getMessage()
+            );
+            return 0.0;
+        }
+    }
+
+    /**
+     * Retrieve MAXSCORE default values for multiple items
+     *
+     * This method retrieves the maximum achievable score for each item
+     * by processing each URI individually. While this requires N file reads
+     * (one qti.xml per item), it enables a single HTTP request instead of N
+     * separate API calls from the frontend.
+     *
+     * @param array $itemUris
+     * @return array<string, float>
+     * @throws Exception
+     */
+    public function getItemsMaxScores(array $itemUris): array
+    {
+        $itemUris = array_filter($itemUris, 'is_string');
+
+        if (empty($itemUris)) {
+            return [];
+        }
+
+        $result = [];
+
+        try {
+            /** @var Service $qtiService */
+            $qtiService = $this->getServiceLocator()->get(Service::class);
+        } catch (Exception $e) {
+            common_Logger::e('Failed to retrieve QTI service: ' . $e->getMessage());
+            throw new Exception('QTI service unavailable: ' . $e->getMessage());
+        }
+
+        foreach ($itemUris as $uri) {
+            try {
```

Comment:
I think it can be reused from the getItemMaxScore to retrieve duplication of code